### DISCUSSION TITLE
TARGET=nrf52840 does not print %f

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2703] [Q&A] [answered]
When compiling for the `nrf52840` platform, it does not print `%f`, for example:

````
#define FAILURE_PROBABILITY 0.10

LOG_INFO("   Failure probability = %f\n", FAILURE_PROBABILITY);
````
Output:
> INFO: Controller]    Failure probability =

**Comment by pjonsson** [comment, marked as answer]
This is the issue that dbg-io doesn't have any code for printing floats/oubles/etc, which you also noticed in Cooja.

I don't know if there's an easy way to use the printf in the libc shipped by ARM with the compiler, but you could remove sprintf from Contiki-NG source code and you should get the sprintf from the libc. The buffer passed to sprintf could then be passed to printf/puts for output. It's not pretty, but it would save you from the effort to figure out how to get the libc-printf to output to the serial port.
